src/integrations/prefect-aws/prefect_aws/experimental/bundles/upload.py

```diff
@@ -85,12 +86,18 @@ def _upload_bundle_to_s3(
         else None
     )
 
-    return upload_bundle_to_s3(
-        local_filepath=local_filepath,
-        bucket=bucket,
-        key=key,
-        aws_credentials_block_name=block_name,
-    )
+    try:
+        return upload_bundle_to_s3(
+            local_filepath=local_filepath,
+            bucket=bucket,
+            key=key,
+            aws_credentials_block_name=block_name,
+        )
+    except Exception as e:
+        print(
+            f"Failed to upload bundle to S3 ({type(e).__name__}): {e}", file=sys.stderr
+        )
+        raise typer.Exit(code=1)
```

Comment:
i considered commenting about using `exit_with_error` but then decided I like how it is better. just thinking out loud about the long term future of `typer`